components/script/dom/gpudevice.rs

```diff
@@ -73,6 +90,13 @@ pub struct GPUDevice {
     label: DomRefCell<Option<DOMString>>,
     device: webgpu::WebGPUDevice,
     default_queue: Dom<GPUQueue>,
+    error_scopes: DomRefCell<HashMap<u64, ErrorScopeInfo>>,
+    scope_stack: DomRefCell<Vec<u64>>,
+    next_scope_id: Cell<u64>,
+    #[ignore_malloc_size_of = "promises are hard"]
+    lost_promise: DomRefCell<Option<Rc<Promise>>>,
+    #[ignore_malloc_size_of = "defined in webgpu"]
+    bind_group_layouts: DomRefCell<Vec<(Vec<wgt::BindGroupLayoutEntry>, Dom<GPUBindGroupLayout>)>>,
```

Comment:
should this be a hashmap instead?

Reply:
We'll have to Implement Hash for wgt::BindGroupLayoutEntry for that.